user-auth-api.py

```
@@ -0,0 +1,108 @@
+        })
+    
+    # User enumeration vulnerability
+    return jsonify({'error': f'User {username} not found'}), 401
+
+@app.route('/reset_password', methods=['POST'])
+def reset_password():
+    email = request.form.get('email')
+    new_password = request.form.get('password')
+    
+    # Weak password hashing
+    hashed = hashlib.md5(new_password.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)